src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -1304,6 +1305,44 @@ def create_or_replace_view(
             source_plan,
         )
 
+    def find_table_function_in_sql_tree(self, plan: SnowflakePlan):
+        from snowflake.snowpark._internal.analyzer.select_statement import (
+            SelectTableFunction,
+            Selectable,
+        )
+
+        if isinstance(plan, SelectTableFunction) and isinstance(
+            plan.snowflake_plan.source_plan, TableFunctionJoin
+        ):
+            if (
+                plan.snowflake_plan.source_plan.right_cols == ["*"]
+                and len(plan.snowflake_plan.children_plan_nodes) == 1
+            ):
+                child_plan = plan.snowflake_plan.children_plan_nodes[0]
+                if isinstance(child_plan, Selectable):
+                    child_plan = child_plan.snowflake_plan
+                plan.snowflake_plan.source_plan.right_cols = (
+                    plan.snowflake_plan.quoted_identifiers[
+                        len(child_plan.quoted_identifiers) :
+                    ]
```

Comment:
I guess this is to flatten "*" to all identifiers.
I have a dumb question, what does `plan.snowflake_plan.quoted_identifiers` contain and why we are only selecting elements after `len(child_plan.quoted_identifiers)`